src/main/java/org/broadinstitute/hellbender/utils/read/GoogleGenomicsReadToGATKReadAdapter.java

```diff
@@ -703,4 +733,105 @@ private void workAroundHttpClientCloneBug() {
             alignment.setCigar(new ArrayList<>(alignment.getCigar()));
         }
     }
+
+    /**
+     * Return a SAM string representation of this read.
+     */
+    private String createSAMString() {
+        final String SAM_FIELD_SEPARATOR = "\t";
+        final String SAM_ATTRIBUTE_SEPARATOR = ":";
+
+        final String contigName = getContig();
+        final StringBuilder sb = new StringBuilder();
+
+        sb.append(getName());
+        sb.append(SAM_FIELD_SEPARATOR);
+        sb.append(Integer.toString(ReadUtils.getSAMFlagsForRead(this)));
+        sb.append(SAM_FIELD_SEPARATOR);
+        sb.append(null == contigName ? SAMRecord.NO_ALIGNMENT_REFERENCE_NAME : contigName);
+        sb.append(SAM_FIELD_SEPARATOR);
+        sb.append(Integer.toString(getStart()));
+        sb.append(SAM_FIELD_SEPARATOR);
+        sb.append(Integer.toString(getMappingQuality()));
+        sb.append(SAM_FIELD_SEPARATOR);
+        sb.append(getCigar().toString());
+        sb.append(SAM_FIELD_SEPARATOR);
+
+        if (!isPaired()) {
```

Comment:
could we pull this block of logic out as a private helper function `getMateContigString` so that it reads more linearly?